cmd/tsbs_generate_queries/uses/common/common.go

```diff
@@ -32,3 +37,30 @@ func NewCore(start, end time.Time, scale int) (*Core, error) {
 func PanicUnimplementedQuery(dg utils.QueryGenerator) {
 	panic(fmt.Sprintf("database (%v) does not implement query", reflect.TypeOf(dg)))
 }
+
+// GetRandomSubsetPerm returns a subset of numItems of a permutation of numbers from 0 to totalNumbers,
+// e.g., 5 items out of 30. This is an alternative to rand.Perm and then taking a sub-slice,
+// which used up a lot more memory and slowed down query generation significantly.
+// The subset of the permutation should have no duplicates and thus, can not be longer that original set
+// Ex.: 12, 7, 25 for numItems=3 and totalItems=30 (3 out of 30)
+func GetRandomSubsetPerm(numItems int, totalItems int) ([]int, error) {
+	if numItems > totalItems {
+		// Cannot make a subset longer than the original set
+		return nil, fmt.Errorf(errMoreItemsThanScale)
+	}
+
+	seen := map[int]bool{}
+	res := []int{}
```

Comment:
since we know the `numItems` we can use a fixed size array and so append doesn't allocate new memory